backend/corpora/dataset_processing/download.py

```diff
@@ -74,6 +76,28 @@ def updater(processing_status_uuid: str, tracker: ProgressTracker, frequency: fl
     """
 
     def _update():
+        with db_session_manager(commit=True) as db:
+            try:
+                curr_status = db.get(DbDatasetProcessingStatus, processing_status_uuid).upload_status
+                if curr_status is UploadStatus.CANCEL_PENDING:
+                    logger.info(f"cancelling the upload for {curr_status.dataset.id}")
+                    # set db to cancelled
+                    status = {
+                        DbDatasetProcessingStatus.upload_progress: 0,
+                        DbDatasetProcessingStatus.upload_status: UploadStatus.CANCELED,
+                        DbDatasetProcessingStatus.upload_message: "Canceled by user",
+                    }
+
+                    dataset = Dataset.get(curr_status.dataset.id)
+                    processing_status_updater(dataset.processing_status.id, status)
+                    # delete from s3
+                    delete_many_from_s3(tracker.artifact_bucket, dataset.id)
+                    delete_many_from_s3(tracker.cellxgene_bucket, dataset.id)
+                    tracker.cancel()
+                elif curr_status is UploadStatus.Canceled:
+                    return
+            except AttributeError:
```

Comment:
Maybe add a comment.  What in the above can raise an attribute error and why is that not a problem.